src/main/kotlin/com/github/quiltservertools/ledger/commands/parameters/SourceParameter.kt

```diff
@@ -19,7 +19,7 @@ class SourceParameter : SimpleParameter<String>() {
         return stringReader.string.substring(i, stringReader.cursor)
     }
 
-    private fun isCharValid(c: Char): Boolean = c in '0'..'9' || c in 'a'..'z' || c in 'A'..'Z' || c == '@' || c == '_'
+    private fun isCharValid(c: Char): Boolean = c in '0'..'9' || c in 'a'..'z' || c in 'A'..'Z' || c == '@' || c == '_' || c == '!'
```

Comment:
Unneeded now I think
```suggestion
    private fun isCharValid(c: Char): Boolean = c in '0'..'9' || c in 'a'..'z' || c in 'A'..'Z' || c == '@' || c == '_' 
```